### PR TITLE
[Backend] User

### PR DESCRIPTION
**Pull request by brixtuyor**
# Related Links
- http://127.0.0.1:8000/
- http://127.0.0.1:8000/login
- http://127.0.0.1:8000/register
- http://127.0.0.1:8000/dashboard
- http://127.0.0.1:8000/lessons
- https://app.asana.com/0/1202403928837322/1202499004499936/f

# Description
- Overhauled most of the functionalities and features
- Transitioned from `Webpack` to `Vite` alongside `Breeze` and `Inertia`

# Notes
- Most of the Laravel dependencies were modified, please run `composer update` and `composer install` right after
- Most of the Node dependencies were added and/or removed, please run `npm install`
- Added extra instructions on how to spin up the development server in `README.md `
- Previous commits were stashed

# Screenshots
image
image
image
image

### CODE REVIEW

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 1]
```diff
@@ -16,12 +16,11 @@ public function up()
         Schema::create('users', function (Blueprint $table) {
             $table->id();
             $table->string('firstname');
-            $table->string("lastname");
+            $table->string('lastname');
             $table->string('email')->unique();
             $table->string('role')->default('basic');
-            $table->string('avatar_url')->nullable();
+            $table->text('avatar_url')->default('https://scontent.fmnl13-2.fna.fbcdn.net/v/t39.30808-6/290593310_184505864007847_1592010923132526048_n.jpg?_nc_cat=107&ccb=1-7&_nc_sid=0debeb&_nc_ohc=g5nQbRYDK4AAX8Cv-nI&_nc_ht=scontent.fmnl13-2.fna&oh=00_AT_BID6Ntxu6uZeGwyAGPpHY-_PFs8IFH-4bIjRzc8zcAg&oe=62BFFD5E');
```
Comment (roseaugusto) [inline database/migrations/2014_10_12_000000_create_users_table.php]: Don't directly edit on migration file, this is bad practice. Make sure to make new migration if there are changes in the models so it won't affect possible production functionality.

**Review by roseaugusto — COMMENTED** [review]
Please update the base url here. currently it's not dynamic. it should come from `.env`

image

**Comment by brixtuyor** [comment]
> Please update the base url here. currently it's not dynamic. it should come from `.env`
> 
> image

Fixed
image
image

File: database/migrations/2014_10_12_000000_create_users_table.php (hunk 1 above)
Comment (brixtuyor) [reply database/migrations/2014_10_12_000000_create_users_table.php]: Removed the default url for `avatar_url`
image

**Pull request merged**